source/kernel/src/servers.rs

```diff
@@ -0,0 +1,100 @@
+//! Servers
+//!
+//! This module contains servers that may be useful for systems
+
+use core::time::Duration;
+
+use crate::{
+    drivers::serial_mux::{PortHandle, WellKnown},
+    tracing, Kernel,
+};
+
+//
+// Sermux Loopback
+//
+
+/// Loopback Settings
+#[derive(Debug, Clone)]
+pub struct SermuxLoopbackSettings {
+    /// Port number. Defaults to [WellKnown::Loopback]
+    pub port: u16,
+    /// Buffer size, in bytes. Defaults to 128
+    pub buffer_size: usize,
+}
+
+impl Default for SermuxLoopbackSettings {
+    fn default() -> Self {
+        Self {
+            port: WellKnown::Loopback as u16,
+            buffer_size: 128,
+        }
+    }
+}
+
+/// Spawns a loopback server
+///
+/// Listens to all input from the given port, and echos it back
+#[tracing::instrument(skip(kernel))]
+pub async fn sermux_loopback(kernel: &'static Kernel, settings: SermuxLoopbackSettings) {
+    let SermuxLoopbackSettings { port, buffer_size } = settings;
+    tracing::debug!("initializing SerMux loopback...");
+    let p0 = PortHandle::open(kernel, port, buffer_size).await.unwrap();
+    tracing::info!("SerMux Loopback running!");
+
+    loop {
+        let rgr = p0.consumer().read_grant().await;
+        let len = rgr.len();
+        tracing::trace!("Loopback read {len}B");
+        p0.send(&rgr).await;
+        rgr.release(len);
+    }
+}
+
+//
+// Sermux Hello
+//
+
+/// Hello Server Settings
+#[derive(Debug, Clone)]
+pub struct SermuxHelloSettings {
```

Comment:
same comment as above; perhaps this ought to have a "no construct without default" field?